File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: list[int]) -> int:
        n = len(nums)
        i = 0

        while i<n:
            x = nums[i]
            if 1 <= x <= n and nums[i] != nums[x-1]:
                nums[i], nums[x-1] = nums[x-1], nums[i]
            else:
                i += 1

        for i in range(n):
            if nums[i] != i+1:
                return i+1
        
        return n+1
        

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int n = nums.length;
        int i = 0;

        while (i < n) {
            int x = nums[i];
            
            if ( x >= 1 && x <=n && nums[i] != nums[x-1]) {
                int temp = nums[i];
                nums[i] = nums[x-1];
                nums[x-1] = temp;

            }
            else {
                i++ ;
            }
        }

        for (i =0; i <nums.length; i++) {
            if (nums[i] != i+1) {
                return(i+1);
            }
        }

        return(n+1);
        



    }
}